docs/api.yaml

```diff
@@ -0,0 +1,88 @@
+openapi: "3.0.0"
+info:
+  version: 1.0.0
+  title: CartBooking API
+  license:
+    name: GPLv3
+servers:
```

Comment:
I am missing authentication here. I was thinking that maybe JWT can be a good way to interact with JS clients. What do you think @johnrivelt?

Reply:
Yes, JSON token communication for auth would be good.

Something like
```angularjs
$http.get("/oauth/v2/token", {params:{
    'client_id':'{{ client.id }}_{{ client.random_id}}',
    'client_secret':'{{ client.secret }}',
    'grant_type':'password',
    'redirect_uri':'{{ redirect_uri }}',
    'username':'admin',
    'password':'admin'
}})

$http.get("/api/v1/bookings", {
    headers:  {
        'Authorization': 'Bearer '+ res.data.access_token
    }})
    .then(function success(res) {
        console.log(res);
        $scope.items = res.data;
    }, function fail(res) {
        console.log(res);
    });
```

Reply:
hey @johnrivelt tonight I will be able to spend time on the site.
I would like to know why do you think is a good idea to use OAuth.

> OAuth is an open standard for access delegation, commonly used as a way for Internet users to grant websites or applications access to their information on other websites but without giving them the passwords.
 From wiki ^

From what I understand, we don't need that. Maybe you have some other use cases in mind. Would you mind to share the ideas? 

From what I understand, we could do this simply by JWT instead. That doesn't require OAuth in place, just symmetric signing on server side, and sharing the token as part of the headers